lib/slather/project.rb

```diff
@@ -51,7 +60,60 @@ def coverage_files
         dedupe(coverage_files)
       end
     end
-    private :coverage_files
+    private :gcov_coverage_files
+
+    def profdata_coverage_files
+      files = profdata_llvm_cov_output.split("\n\n")
+
+      files.map do |source|
+        coverage_file = coverage_file_class.new(self, source)
+        !coverage_file.ignored? ? coverage_file : nil
+      end.compact
+    end
+    private :profdata_coverage_files
+
+    def profdata_coverage_dir
+      if self.scheme
+        Dir["#{build_directory}/**/CodeCoverage/#{self.scheme}"].first
+      else
+        Dir["#{build_directory}/**/#{self.products.first.name}"].first
+      end
+    end
+
+    def binary_file
+      xctest_bundle_file = Dir["#{profdata_coverage_dir}/**/*.xctest"].first
+      if xctest_bundle_file == nil
+        raise StandardError, "No product binary found in #{profdata_coverage_dir}"
+      end
+
+      # Find the matching .app, if any
+      xctest_bundle_file_directory = Pathname.new(xctest_bundle_file).dirname
+
+      app_bundle_file = Dir["#{xctest_bundle_file_directory}/*.app"].first
+      framework_bundle_file = Dir["#{xctest_bundle_file_directory}/*.framework"].first
+      if app_bundle_file != nil
+        app_bundle_file_name_noext = Pathname.new(app_bundle_file).basename.to_s.gsub(".app", "")
+        "#{app_bundle_file}/#{app_bundle_file_name_noext}"
+      elsif framework_bundle_file != nil
+          framework_bundle_file_name_noext = Pathname.new(framework_bundle_file).basename.to_s.gsub(".framework", "")
+          "#{framework_bundle_file}/#{framework_bundle_file_name_noext}"
+      else
+        xctest_bundle_file_name_noext = Pathname.new(xctest_bundle_file).basename.to_s.gsub(".xctest", "")
+        "#{xctest_bundle_file}/#{xctest_bundle_file_name_noext}"
+      end
+    end
+    private :binary_file
+
+    def profdata_llvm_cov_output
+      profdata_coverage_dir = self.profdata_coverage_dir
+      if profdata_coverage_dir == nil || (coverage_profdata = Dir["#{profdata_coverage_dir}/**/Coverage.profdata"].first) == nil
+        raise StandardError, "No Coverage.profdata files found. Please make sure the \"Code Coverage\" checkbox is enabled in your scheme's Test action or the build_directory property is set."
+      end
+      xcode_path = `xcode-select -p`.strip
+      llvm_cov_command = File.join(xcode_path, "Toolchains/XcodeDefault.xctoolchain/usr/bin/llvm-cov show -instr-profile #{coverage_profdata} #{binary_file}")
+      `#{llvm_cov_command}`
```

Comment:
@pietbrauer Nice job on putting this together. I've extracted some of the concepts here and added them directly into Codeocov's bash uploader, as seen below:

```bash
profdata=$(find ~/Library/Developer/Xcode/DerivedData -name 'Coverage.profdata' | head -1)
if [ -f "$profdata" ];
then
  _dir=$(dirname "$profdata")
  for _type in app framework xctest
  do
    _file=$(find "$_dir" -name "*.$_type" | head -1)
    if [ "$_file" != "" ];
    then
      _proj=${_file##*/}
      _proj=${_proj%."$_type"}
      xcrun llvm-cov show -instr-profile "$profdata" "$_file/$_proj" > "$_type.coverage.txt" || true
    fi
  done
fi
```
> Extracted from [codecov/codecov-bash#L367](https://github.com/codecov/codecov-bash/blob/b083db215bf29d5eb9b49c2ab10f3fd232400d72/codecov#L367-L382)

So far a couple customers of Codecov are using this successfully. However, we are seeing some projects with 0% coverage. One customer said they fixed it by enabling  **Gather code coverage** in `xcodebuild` from within xcode. 

Do you know of a reason why coverage would be 0%?

> PS I'm not a xcode developer so my apologies if I'm not accurate with my description, they are quotes from customer support.